### ISSUE TITLE
Proposal: advanced search syntax

### ISSUE THREAD

**Issue by stereobooster** [activerecord-hackery/ransack#1364]
Github has [search syntax](https://docs.github.com/en/search-github/getting-started-with-searching-on-github/understanding-the-search-syntax). Can we do the same with ransack? For example, query "x:1 y:>2" converted to `{ x_eq: 1, y_gt: 2}`.

I built prototype for this https://github.com/stereobooster/search_syntax. WDYT?

**Comment by scarroll32** [comment]
@stereobooster this looks like a really interesting idea, let me look into it.

**Comment by ericenns** [comment]
@stereobooster that is a really great idea, are you still using your prototype?

**Comment by stereobooster** [comment]
@ericenns yes, it is used in one project. To be fair I never had to touch it again after implementation

**Comment by scarroll32** [comment]
> [!NOTE]
> Comment from **Claude** (Claude Code), acting on behalf of @scarroll32.

A good idea, and it already exists as your `search_syntax` gem, which turns a query string into a Ransack params hash without Ransack needing to know about it. That is the right layering: Ransack stays a params-to-SQL library and query languages live on top. Closing as out of scope; it will be linked from the External resources docs page. For questions and design discussions please use the [Discussions forum](https://github.com/activerecord-hackery/ransack/discussions); the issue tracker is kept for reproducible bugs.
